core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -74,7 +74,7 @@
    * Maximum supported data page size (in bytes). In principle, the maximum addressable page size is
    * (1L &lt;&lt; OFFSET_BITS) bytes, which is 2+ petabytes. However, the on-heap allocator's
    * maximum page size is limited by the maximum amount of data that can be stored in a long[]
-   * array, which is (2^32 - 1) * 8 bytes (or 16 gigabytes). Therefore, we cap this at 16 gigabytes.
+   * array, which is (2^31 - 1) * 8 bytes (or 16 gigabytes). Therefore, we cap this at 16 gigabytes.
```

Comment:
(2^31-1)*8 = 17179869176 Bytes = 17179869176/1024/1024/1024 ≈ 16GB, "or about 16GB" should not wrong.